### PR TITLE
Add support for nv_ray_tracing final 

### PR DESCRIPTION
**Pull request by alelenv**
GLSL Spec : 
https://gitlab.khronos.org/GLSL/GLSL/merge_requests/30

SPV Spec:
https://gitlab.khronos.org/spirv/spirv-extensions/merge_requests/124

@johnkslang for review
@dgkoch as cc

### CODE REVIEW

File: SPIRV/doc.cpp [hunk 1]
```diff
@@ -889,7 +892,7 @@ const char* CapabilityString(int info)
     case CapabilityShaderStereoViewNV:              return "ShaderStereoViewNV";
     case CapabilityPerViewAttributesNV:             return "PerViewAttributesNV";
     case CapabilityGroupNonUniformPartitionedNV:    return "GroupNonUniformPartitionedNV";
-    case CapabilityRaytracingNVX:                   return "RaytracingNVX";
+    case CapabilityRaytracingNV:                    return "RaytracingNV";
```
Comment (dgkoch) [inline SPIRV/doc.cpp]: Should be RayTracingNV

Comment (alelenv) [reply SPIRV/doc.cpp]: Done. Thanks for catching this

**Comment by alelenv** [comment]
@johnkslang 
John, The appveyor builds are failing due to weird host compiler failures
C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\xmemory0(592): error C3859: virtual memory range for PCH exceeded; please recompile with a command line option of '-Zm124' or greater [C:\projects\glslang\build\External\spirv-tools\source\opt\

Rest complains about googletests build failures which are identical for other pull request already merged
https://ci.appveyor.com/project/Khronoswebmaster/glslang/builds/19826579

Can you assist? Thanks

**Comment by alelenv** [comment]
@dneto0 @dj2 as SPIRV tools owners
I see following in build log for spirv opt in appveyor which I believe is **latent issue** and unrelated to my change

C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\xmemory0(592): error C3859: virtual memory range for PCH exceeded; please recompile with a command line option of '-Zm124' or greater [C:\projects\glslang\build\External\spirv-tools\source\opt\SPIRV-Tools-opt.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\xmemory0(592): fatal error C1076: compiler limit : internal heap limit reached; use /Zm to specify a higher limit [C:\projects\glslang\build\External\spirv-tools\source\opt\SPIRV-Tools-opt.vcxproj]

I believe this needs an upgrade to CMakeLists.txt with something like below
set_source_files_properties(pch.cpp PROPERTIES COMPILE_FLAGS "/Ycpch.h **/Zm1000** /Fp${PCH_NAME}" OBJECT_OUTPUTS "${PCH_NAME}")

**Comment by johnkslang** [comment]
I have reviewed, and have no issues with the details.

What is the current status on making the tests pass?

**Comment by jeffbolznv** [comment]
The build failure appears to be triggered by the PCH stuff I added to spirv-tools recently. The error is:

```
C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\xmemory0(592): error C3859: virtual memory range for PCH exceeded; please recompile with a command line option of '-Zm124' or greater [C:\projects\glslang\build\External\spirv-tools\source\opt\SPIRV-Tools-opt.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\xmemory0(592): fatal error C1076: compiler limit : internal heap limit reached; use /Zm to specify a higher limit [C:\projects\glslang\build\External\spirv-tools\source\opt\SPIRV-Tools-opt.vcxproj]
```

I can't reproduce this with Visual Studio 2015, and we haven't had any problems with PCH in CTS either, so this seems like it might be a VS2013-ism. I guess we should follow the error's advice and increase this memory limit (say, -Zm200?), but it would be nice to verify that it works before pushing that to spirv-tools.

**Comment by jeffbolznv** [comment]
Heh, I should have read AL's comment from earlier. I'll go ahead and make that change in spirv-tools, though it will be essentially untested.

**Comment by jeffbolznv** [comment]
I created https://github.com/KhronosGroup/SPIRV-Tools/pull/2023 for this. AL, please keep an eye on it and pull the updated commit hash whenever it is merged.

**Comment by alelenv** [comment]
Squashed to single commit with updated known_good. All builds are now passing and is ready to merge

**Pull request merged**